src/asmcnc/apps/drywall_cutter_app/material_setup_popup.py

```diff
@@ -0,0 +1,428 @@
+        self.cut_depth_warning.text = self.cut_depth_warning_soft_limit
+
+    def on_touch(self):
+        for text_input in self.text_inputs:
+            text_input.focus = False
+            if text_input.text != '':
+                try:
+                    text_input.text = str(float(text_input.text))
+                except:
+                    pass
+            else:
+                text_input.text = str(float(0))
+
+    def on_checkbox_active(self):
+        if self.auto_pass_checkbox.active:
+            self.depth_per_pass.disabled = True
+            self.depth_per_pass.text = self.depth_per_pass.hint_text
+            self.depth_per_pass.hint_text = ''
+        else:
+            self.depth_per_pass.disabled = False
+            self.depth_per_pass.hint_text = self.depth_per_pass.text
+            self.depth_per_pass.text = ''
+
+    def update_text(self, instance, value):
+        cutter_max_depth_total = self.dwt_config.active_cutter.max_depth_total
+        # Calculate total cut depth and handle inputs
+        if instance == self.material_thickness or instance == self.bottom_offset:
+            try:
+                # Update warning label text
+                if self.soft_limit_total_cut_depth < cutter_max_depth_total:
+                    max_cut_depth = self.soft_limit_total_cut_depth
+                    self.cut_depth_warning.text = self.cut_depth_warning_soft_limit
+                else:
+                    max_cut_depth = cutter_max_depth_total
+                    self.cut_depth_warning.text = self.cut_depth_warning_cutter_max
+
+                material_thickness = 0 if self.material_thickness.text == '' or self.material_thickness.text == '-' else float(
+                    self.material_thickness.text)
+                bottom_offset = 0 if self.bottom_offset.text == '' or self.bottom_offset.text == '-' else float(
+                    self.bottom_offset.text)
+
+                # Make sure material thickness is never negative
+                if material_thickness < 0:
+                    self.material_thickness.text = '0'
+                    material_thickness = 0
+
+                # Make sure bottom offset is never bigger than material thickness
+                if abs(bottom_offset) > material_thickness:
+                    if bottom_offset < 0:
+                        if self.material_thickness.text == '':
+                            self.bottom_offset.text = ''
+                            bottom_offset = material_thickness
+                        else:
+                            self.bottom_offset.text = '-' + self.material_thickness.text
+                            bottom_offset = -material_thickness
+
+                self.total_cut_depth.text = str(material_thickness + bottom_offset)
+
+                if float(self.total_cut_depth.text) > max_cut_depth:
+                    self.float_layout.add_widget(self.cut_depth_warning)
```

Comment:
What if I make the total_cut_depth.text > max_cut_depth, then this function runs, then I do it again and set it to another value which is also > max_cut_depth, will it add the widget twice?